pkg/gfx/program.go

```diff
@@ -0,0 +1,80 @@
+package gfx
+
+import (
+	"fmt"
+
+	"github.com/go-gl/gl/v2.1/gl"
+	"github.com/gregjohnson2017/tabula-editor/pkg/log"
+)
+
+type Program struct {
+	id uint32
+}
+
+// ErrProgramLink indicates that a program failed to link
+const ErrProgramLink log.ConstErr = "failed to compile shader"
+
+// CreateShaderProgram compiles a vertex and fragment shader,
+// attaches them to a new shader program and returns its ID.
+func NewProgram(shaders []Shader) (Program, error) {
```

Comment:
Using variadic parameters here would be easier to use.

Reply:
done